File: media/textsamples/sample.lua
-- sample.lua — Lua syntax-highlighting sample for the UltraCanvas demo.
local Vector = {}
Vector.__index = Vector

function Vector.new(x, y)
    return setmetatable({ x = x or 0, y = y or 0 }, Vector)
end

function Vector:lengthSquared()
    return self.x * self.x + self.y * self.y
end

function Vector.__add(a, b)
    return Vector.new(a.x + b.x, a.y + b.y)
end

local points = {
    Vector.new(1, 2),
    Vector.new(3, 4),
    Vector.new(-5, 0.5),
}

table.sort(points, function(a, b)
    return a:lengthSquared() < b:lengthSquared()
end)

for i, p in ipairs(points) do
    print(string.format("%d: (%.1f, %.1f)", i, p.x, p.y))
end

local sum = points[1] + points[2]
print("sum = (" .. sum.x .. ", " .. sum.y .. ")")

File: media/textsamples/sample.r
# sample.r — R syntax-highlighting sample for the UltraCanvas demo.
# Summarise a small data frame of measurements per group.

measurements <- data.frame(
  group = factor(c("a", "b", "a", "c", "b", "c", "a")),
  value = c(2.5, 3.75, 1.25, NA, 4.0, 6.5, 3.0)
)

describe <- function(x, digits = 2) {
  x <- x[!is.na(x)]
  if (length(x) == 0) return(NULL)
  list(n = length(x), mean = round(mean(x), digits), sd = round(sd(x), digits))
}

summaries <- lapply(split(measurements$value, measurements$group), describe)

for (name in names(summaries)) {
  s <- summaries[[name]]
  if (is.null(s)) next
  cat(sprintf("%s: n=%d mean=%.2f sd=%.2f\n", name, s$n, s$mean, s$sd))
}

above <- subset(measurements, value > 3 & !is.na(value))
print(head(above[order(-above$value), ], 3))

squares <- sapply(1:5, function(i) i^2)
stopifnot(sum(squares) == 55L, all(squares %in% c(1, 4, 9, 16, 25)))
